keyboards/contra/keymaps/bramver/keymap.c

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2018 darm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum emoji_map {
+  UNAM, // unamused 😒
+  HEYE, // smiling face with heart shaped eyes 😍
+  OK, // ok hand sign 👌
+  SMIR, // smirk 😏
+  PRAY, // pray 🙏
+  CELE, // celebration 🙌
+  COOL, // smile with sunglasses 😎
+  EYES, // eyes
+  THNK, // BIG THONK
+  NAIL, // Nailcare
+  SOS, // Vuile sos
+  REDB, // Red B
+  HNDR, // 100
+  MONY,
+  FIRE,
+  CAR,
+  BUTT,
+  BNIS,
+  CUM,
+  CLAP,
+  TRIU, // Fart from nose
+  SCRM,
+  VOMI,
+  DTIV, // Detective
+  EXPL, // Brainsplosion
+  HAIR, // Haircut
+  DANC, // Salsa dancer
+  STRN, // Stronk
+  LEFT, // Point Left
+  RGHT, // Point Right
+};
+
+const uint32_t PROGMEM unicode_map[] = {
+  [UNAM] = 0x1F612,
+  [HEYE] = 0x1f60d,
+  [OK]   = 0x1F44C,
+  [SMIR] = 0x1F60F,
+  [PRAY] = 0x1F64F,
+  [CELE] = 0x1F64C,
+  [COOL] = 0x1F60E,
+  [EYES] = 0x1F440,
+  [THNK] = 0x1F914,
+  [NAIL] = 0x1F485,
+  [SOS] = 0x1F198,
+  [REDB] = 0x1F171,
+  [HNDR] = 0x1F4AF,
+  [MONY] = 0x1F480,
+  [FIRE] = 0x1F525,
+  [CAR] = 0x1F697,
+  [BUTT] = 0x1F351,
+  [BNIS] = 0x1F346,
+  [CUM] = 0x1F4A6,
+  [CLAP] = 0x1F44F,
+  [TRIU] = 0x1F624,
+  [SCRM] = 0x1F631,
+  [VOMI] = 0x1F92E,
+  [DTIV] = 0x1F575,
+  [EXPL] = 0x1F92F,
+  [HAIR] = 0x2640,
+  [DANC] = 0x1F483,
+  [STRN] = 0x1F4AA,
+  [LEFT] = 0x1F448,
+  [RGHT] = 0x1F449,
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
+#define XXXXXXX KC_NO
+
+// Layer shorthand
+#define _BASE     0
+#define _LOWER    1
+#define _EMOJI    2
+#define _MOUSE    3
+
+#define SP_MSE LT(_MOUSE, KC_SPC)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = {
```

Comment:
Could you reformat the layers so that it uses the `LAYOUT` format?

Eg:
https://github.com/qmk/qmk_firmware/blob/4e9fb1d0d93db07c9934f64f6c8fe7a0dce51464/keyboards/contra/keymaps/dana/keymap.c#L25-L30

Reply:
There are a few reasons actually.  

What you had before is the "raw matrix".  That's "okay", but not ideal.    If a revision is added and the matrix changed, this causes problems and would require you to change it anyways.  This is actually an issue that we had recently with the Planck and Preonic boards.  The matrix is very different on them, so the raw matrix is not valid for many of the keymaps.

Additionally, if you ever want to port this to another board, like the planck, lets split, fractal, zlant, or other similarly laid out boards, using the `LAYOUT` makes it "drag and drop" easy, as many of those boards use very different matrices, as well. 

So, because of these reasons, we try to have people use the layout "macros" instead of the raw matrix.

Reply:
Cool that sounds much better indeed. Do other boards have this as well (Quefrency for example), or only the one with a similar layout?  
Because I could go back and update my previous keymaps.

Reply:
Yes, most boards have this.

Here's one of the layout macros for the Quefrency:

https://github.com/qmk/qmk_firmware/blob/fc1e22ea94146d090af5b47a51fcc07cac7e677c/keyboards/quefrency/rev1/rev1.h#L16-L36

Reply:
Okay I'll make sure to update my previous maps then when I have some free time next week :+1:

Reply:
@BramVer Your XD75 keymap seems to be the only one in the QMK repo that needs updating, assuming you don't have other keymaps around that aren't in the repo.

Here's what your XD75 keymap's base layer should look like when you're done:

```c
  [_BASE] = LAYOUT_ortho_5x15(
    KC_GESC , KC_1  , KC_2    , KC_3    , KC_4    , KC_5    , KC_MPRV , KC_MPLY , KC_MNXT , KC_6    , KC_7    , KC_8    , KC_9    , KC_0    , KC_BSPC,
    KC_TAB  , KC_Q  , KC_W    , KC_E    , KC_R    , KC_T    , KC_MINS , KC_EQL  , KC_BSLS , KC_Y    , KC_U    , KC_I    , KC_O    , KC_P    , KC_ENT ,
    MO(3)   , KC_A  , KC_S    , KC_D    , KC_F    , KC_G    , KC_LBRC , KC_MUTE , KC_RBRC , KC_H    , KC_J    , KC_K    , KC_L    , KC_SCLN , KC_QUOT,
    KC_LSFT , KC_Z  , KC_X    , KC_C    , KC_V    , KC_B    , KC_HOME , KC_SLEP , KC_END  , KC_N    , KC_M    , KC_COMM , KC_DOT  , KC_SLSH , KC_RSFT,
    KC_LCTL , MO(4) , KC_LALT , KC_LGUI , KC_SPC  , KC_SPC  , MO(1)   , KC_DEL  , MO(2)   , KC_SPC  , KC_LEFT , KC_DOWN , KC_UP   , KC_RGHT , KC_RGUI
  ),
```

Note that `KC_RGUI` doesn't have a comma after it. Watch out for this, because it's a build-breaking error if you have a comma there (the keymap will fail to compile and error out).

Reply:
Yeah I noticed that when compiling this keymap, one of the most straightforward compile errors I've come across so far.  
Thanks for checking tho.  
I'm working on a CRKBD layout atm, but I've gotta finish building the thing first.